federation/pkg/federation-controller/sync/controller.go

```diff
@@ -415,14 +414,25 @@ func (s *FederationSyncController) objFromCache(kind, key string) (pkgruntime.Ob
 }
 
 // delete deletes the given resource or returns error if the deletion was not complete.
-func (s *FederationSyncController) delete(obj pkgruntime.Object, kind string, namespacedName types.NamespacedName) error {
-	glog.V(3).Infof("Handling deletion of %s %q", kind, namespacedName)
+func (s *FederationSyncController) delete(obj pkgruntime.Object, kind string, qualifiedName federatedtypes.QualifiedName) error {
+	glog.V(3).Infof("Handling deletion of %s %q", kind, qualifiedName)
+
+	// Perform pre-deletion cleanup for the namespace adapter
```

Comment:
I'm on the fence whether this should be a special case like this or a function that you pass in.

It seems like the namespace controller is special in this respect, so I suppose having this be a special case is OK.